### PR TITLE
Convert Notification support classes to kotlin.

### PR DESCRIPTION
**Pull request by dbrant**

### CODE REVIEW

File: app/src/main/java/org/wikipedia/notifications/NotificationPollBroadcastReceiver.kt [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+        @JvmStatic
+        fun markRead(wiki: WikiSite, notifications: List<Notification?>, unread: Boolean) {
+            val idListStr = notifications.joinToString("|")
+            val editTokenClient = CsrfTokenClient(wiki, WikipediaApp.getInstance().wikiSite)
+            editTokenClient.request(object : CsrfTokenClient.DefaultCallback() {
+                @SuppressLint("CheckResult")
+                override fun success(token: String) {
+                    ServiceFactory.get(wiki).markRead(token, if (unread) null else idListStr, if (unread) idListStr else null)
+                            .subscribeOn(Schedulers.io())
+                            .subscribe({ }) { t: Throwable -> L.e(t) }
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/notifications/NotificationPollBroadcastReceiver.kt:252]: Explicit type specification can be removed.

File: app/src/main/java/org/wikipedia/notifications/NotificationPresenter.kt [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+object NotificationPresenter {
+    private const val CHANNEL_ID = "MEDIAWIKI_ECHO_CHANNEL"
+
+    fun showNotification(context: Context, n: Notification, wikiSiteName: String) {
+        @DrawableRes var iconResId = R.drawable.ic_speech_bubbles
+        @ColorRes var iconColor = R.color.accent50
+
+        val builder = getDefaultBuilder(context, n.id(), n.type())
+        val title: String = fromHtml(if (n.contents != null) n.contents!!.header else "").toString()
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/notifications/NotificationPresenter.kt:35]: Should we use `StringUtil.fromHtml()`?

File: app/src/main/java/org/wikipedia/notifications/NotificationPresenter.kt [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+            } else {
+                addAction(context, builder, n.contents!!.links!!.primary, n)
+            }
+        }
+        if (n.contents != null && n.contents!!.links != null && n.contents!!.links!!.secondary != null && n.contents!!.links!!.secondary!!.size > 0) {
+            addAction(context, builder, n.contents!!.links!!.secondary!![0], n)
+        }
+        if (n.contents != null && n.contents!!.links != null && n.contents!!.links!!.secondary != null && n.contents!!.links!!.secondary!!.size > 1) {
+            addAction(context, builder, n.contents!!.links!!.secondary!![1], n)
+        }
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/notifications/NotificationPresenter.kt:49]: Would it be possible to do something like:

```
n.contents?.links?.let {
    it.primary?.let { primary ->
       ...
    }
}

```

File: app/src/main/java/org/wikipedia/notifications/NotificationPresenter.kt [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+                addIntentExtras(Intent(Intent.ACTION_VIEW, Uri.parse(link!!.url)), n.id(), n.type()), 0)
+        val labelStr: String = if (link.tooltip.isNotEmpty()) {
+            fromHtml(link.tooltip).toString()
+        } else {
+            fromHtml(link.label).toString()
+        }
+        builder.addAction(0, labelStr, pendingIntent)
+    }
+
+    private fun addActionForTalkPage(context: Context, builder: NotificationCompat.Builder, link: Notification.Link?, n: Notification) {
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/notifications/NotificationPresenter.kt:142]: Will the `Notification.Link` have the chance to be `null`?

**Review by cooltey — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/wikipedia/notifications/NotificationPresenter.kt (hunk 2 above)
Comment (dbrant) [reply app/src/main/java/org/wikipedia/notifications/NotificationPresenter.kt]: ugh, more quirks from the automatic conversion.

File: app/src/main/java/org/wikipedia/notifications/NotificationPresenter.kt (hunk 3 above)
Comment (dbrant) [reply app/src/main/java/org/wikipedia/notifications/NotificationPresenter.kt:48]: nice

**Pull request merged**